### ISSUE TITLE
Bug In "Art Generation with Neural Style Transfer"

### ISSUE THREAD

**Issue by niunuinui** [furkanu/deeplearning.ai-pytorch#2]
Hi,  I learn a lot from your code! Thanks a lot!

I think there is a bug in the `save_image` function of `Art Generation with Neural Style Transfer` code. Since [.detach() share the same memory,](https://pytorch.org/docs/stable/autograd.html#torch.Tensor.detach) we should clone the input image first or the change will apply to the original image and mess the training.

Bellow is the fixed code for `save_image`
```
def save_image(path, imageO):
    # Un-normalize the image so that it looks good
    image=imageO.clone().detach()
    image = image.cpu().numpy().transpose(0, 2, 3, 1)
    image += CONFIG.MEANS
    image = np.clip(image[0], 0, 255).astype('uint8')

```

**Comment by r1ght0us** [comment]
> Hi, I learn a lot from your code! Thanks a lot!
> 
> I think there is a bug in the `save_image` function of `Art Generation with Neural Style Transfer` code. Since [.detach() share the same memory,](https://pytorch.org/docs/stable/autograd.html#torch.Tensor.detach) we should clone the input image first or the change will apply to the original image and mess the training.
> 
> Bellow is the fixed code for `save_image`
> 
> ```
> def save_image(path, imageO):
>     # Un-normalize the image so that it looks good
>     image=imageO.clone().detach()
>     image = image.cpu().numpy().transpose(0, 2, 3, 1)
>     image += CONFIG.MEANS
>     image = np.clip(image[0], 0, 255).astype('uint8')
> ```

牛逼好兄弟，终于找到原因了！！！！
